build_tools/scripts/generate_release_index.py

```diff
@@ -44,7 +44,8 @@ def get_all(self):
         url = f"https://api.github.com/repos/{self._repo}/releases"
         page = 1
 
-        while True:
+        # GitHub limits API responses to the first 1000 results.
+        while page * self._per_page < 1000:
```

Comment:
Error message:
```
{"message":"Only the first 1000 results are available.","documentation_url":"https://docs.github.com/rest/releases/releases#list-releases"}
```
Sample failed job run: https://github.com/openxla/iree/actions/runs/7979175829/job/21785972004

(We chatted about this on Discord here: https://discord.com/channels/689900678990135345/1062405112292712499/1209653373519921163)